core/src/main/java/org/elasticsearch/action/admin/indices/flush/TransportShardFlushAction.java

```diff
@@ -61,16 +58,16 @@ protected ActionWriteResponse newResponseInstance() {
     }
 
     @Override
-    protected Tuple<ActionWriteResponse, ShardFlushRequest> shardOperationOnPrimary(ClusterState clusterState, PrimaryOperationRequest shardRequest) throws Throwable {
-        IndexShard indexShard = indicesService.indexServiceSafe(shardRequest.shardId.getIndex()).getShard(shardRequest.shardId.id());
-        indexShard.flush(shardRequest.request.getRequest());
+    protected Tuple<ActionWriteResponse, ShardFlushRequest> shardOperationOnPrimary(MetaData metaData, ShardFlushRequest shardRequest) throws Throwable {
```

Comment:
> whereas passing in the metadata hints at using it to validate the request before performing any operations

I'm not sure this hints what you suggest.

To me cluster state is simple  in terms of thinking - you get everything, do what ever you want. No a biggy though. If you feel strongly about this, let's just keep it.